app/controllers/users_controller.rb

```diff
@@ -81,6 +81,11 @@ def changed_username
       render(:change_username) && return
     end
 
+    if @new_login == @user.login
+      flash.now[:error] = t(".user.not_different") 
```

Comment:
Sorry for the extra nitpick on top of the pain of moving the validation, but I think we'll want a different key now that it's here
```suggestion
      flash.now[:error] = t(".new_username_must_be_different") 
```
or possibly
```suggestion
      flash.now[:error] = t(".username.must_be_different") 
```
(It's a bit easier for the translators if the key name is close to the English text when it makes sense/is possible for us to do.)